src/System.Data.Odbc/tests/OdbcParameterTests.cs

```diff
@@ -0,0 +1,268 @@
+            Assert.Equal(str5000.Length, inputLength);
+            Assert.Equal(str8000.Length, outputLength);
+
+            RunTestProcedure("VARCHAR", 8000, str4000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str8000, outputStr);
+            Assert.Equal(str4000.Length, inputLength);
+            Assert.Equal(str8000.Length, outputLength);
+
+            RunTestProcedure("VARCHAR", 8000, str2000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str4000, outputStr);
+            Assert.Equal(str2000.Length, inputLength);
+            Assert.Equal(str4000.Length, outputLength);
+
+            RunTestProcedure("VARCHAR", 8000, str1000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str2000, outputStr);
+            Assert.Equal(str1000.Length, inputLength);
+            Assert.Equal(str2000.Length, outputLength);
+
+            RunTestProcedure("NVARCHAR", 4000, str8000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str4000, outputStr);
+            Assert.Equal(str4000.Length * 2, inputLength); // since NVARCHAR takes 2 bytes per character
+            Assert.Equal(str4000.Length * 2, outputLength);
+
+            RunTestProcedure("NVARCHAR", 4000, str5000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str4000, outputStr);
+            Assert.Equal(str4000.Length * 2, inputLength);
+            Assert.Equal(str4000.Length * 2, outputLength);
+
+            RunTestProcedure("NVARCHAR", 4000, str4000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str4000, outputStr);
+            Assert.Equal(str4000.Length * 2, inputLength);
+            Assert.Equal(str4000.Length * 2, outputLength);
+
+            RunTestProcedure("NVARCHAR", 4000, str2000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str4000, outputStr);
+            Assert.Equal(str2000.Length * 2, inputLength);
+            Assert.Equal(str4000.Length * 2, outputLength);
+
+            RunTestProcedure("NVARCHAR", 4000, str1000, out output, out inputLength, out outputLength);
+            outputStr = output as string;
+            Assert.Equal(str2000, outputStr);
+            Assert.Equal(str1000.Length * 2, inputLength);
+            Assert.Equal(str2000.Length * 2, outputLength);
+        }
+
+        private static void RunTestProcedure(string procDataType, int procDataSize, object v1, out object v2, out int v3, out int v4)
+        {
+            string removeExistingStoredProcSql =
+                "IF OBJECT_ID('testStoredProc', 'P') IS NOT NULL " +
```

Comment:
Can you prefix the StoredProc with a GUID or a timestamp, so that multiple tests can be run against the same server?